app/api/v1/mediator/manifest.py

```diff
@@ -1,53 +1,142 @@
+    if not crypto_hash:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST, detail="manifest hash not valid"
+        )
+    try:
+        with get_repository(CLIENT_ID, DataCollection.MANIFEST) as repository:
+            query_result = repository.get({"manifest_hash": crypto_hash.to_hex()})
+            if not query_result:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail=f"Could not find manifest {manifest_hash}",
+                )
+
+            return ManifestQueryResponse(
+                status=ResponseStatus.SUCCESS,
+                manifest=query_result["manifest"],
+            )
+    except Exception as error:
+        print(sys.exc_info())
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="get manifest failed",
+        ) from error
+
+
+@router.put("", tags=[MANIFEST], status_code=status.HTTP_204_NO_CONTENT)
+def submit_manifest(
+    request: ValidateManifestRequest = Body(...),
+    schema: Any = Depends(get_description_schema),
+) -> ManifestSubmitResponse:
+    """
+    Submit a manifest for storage
+    """
+    manifest, validation = _validate_manifest(request, schema)
+    if not manifest or validation.status == ResponseStatus.FAIL:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST, detail=validation.details
+        )
+
+    try:
+        with get_repository(CLIENT_ID, DataCollection.MANIFEST) as repository:
+            manifest_hash = manifest.crypto_hash().to_hex()
+            print(manifest.to_json_object())
+            keys = repository.set(
+                {"manifest_hash": manifest_hash, "manifest": manifest.to_json_object()}
+            )
+            print(keys)
+            return ManifestSubmitResponse(
+                status=ResponseStatus.SUCCESS, manifest_hash=manifest_hash
+            )
+    except Exception as error:
+        print(sys.exc_info())
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Submit manifest failed",
+        ) from error
+
 
 @router.post("/validate", tags=[MANIFEST], status_code=status.HTTP_204_NO_CONTENT)
-def validate_election_manifest(
+def validate_manifest(
     request: ValidateManifestRequest = Body(...),
     schema: Any = Depends(get_description_schema),
 ) -> BaseValidationResponse:
     """
     Validate an Election manifest for a given election.
     """
 
-    success = ResponseStatus.SUCCESS
-    message = "Election manifest successfully validated"
-    details = None
+    _, response = _validate_manifest(request, schema)
+    return response
+
 
+def _deserialize_manifest(data: object) -> Optional[Manifest]:
+    try:
+        return Manifest.from_json_object(data)
+    except Exception:  # pylint: disable=broad-except
+        # TODO: some sort of information why it failed
+        return None
```

Comment:
we are going to need to add exceptions into the core python repo for this part but it can wait until later